### PR TITLE
Add some test specs

### PR DESCRIPTION
**Pull request by jeanconn**
Add some test specs

### CODE REVIEW

File: test_spec_PY3 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Excludes
+- annie
```
Comment (taldcroft) [inline test_spec_PY3]: Why is annie excluded?

File: test_spec_NONHEAD_REGRESS [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Includes
+acisfp_check
+dpa_check
+dea_check
+psmc_check
+starcheck
+# Excludes
+-*head*
```
Comment (taldcroft) [inline test_spec_NONHEAD_REGRESS]: This glob should be `*_head*` or maybe even `*_head.py`?

File: test_spec_PY3 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Excludes
```
Comment (taldcroft) [inline test_spec_PY3]: Change PY3 to SKA3_HEAD.  This is specifically targeted to run on HEAD machines right?

File: test_spec_PY3_GRETA [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Excludes
```
Comment (taldcroft) [inline test_spec_PY3_GRETA]: `test_spec_SKA3`.  This is the generic one that should pass on any Ska3 installation.

File: test_spec_NONHEAD_REGRESS (hunk 2 above)
Comment (taldcroft) [reply test_spec_NONHEAD_REGRESS]: And rename to `test_spec_SKA3_REGRESS`.  (Making `HEAD` be an opt-in instead of the default).

**Comment by jeanconn** [comment]
I was largely just tossing these here for discussion so this is good.

File: test_spec_NONHEAD_REGRESS (hunk 2 above)
Comment (jeanconn) [reply test_spec_NONHEAD_REGRESS]: What do you mean by making 'HEAD' an opt-in and how did you want to effect that change?

File: test_spec_PY3 (hunk 1 above)
Comment (jeanconn) [reply test_spec_PY3]: Most of the tests were broken up until yesterday's release.  Had a plan to remove the exclude.

File: test_spec_NONHEAD_REGRESS (hunk 2 above)
Comment (taldcroft) [reply test_spec_NONHEAD_REGRESS]: Opt-in meaning that the default tests like `test_spec_SKA3` or `test_spec_SKA3_REGRESS` should pass on any supported platform , while `test_spec_SKA3_HEAD` would require being on HEAD to pass.  The effect-ation is via adding HEAD to the spec file name, though one could certainly imagine some way to do this within the scripts themselves.  But that's a different story.

**Comment by jeanconn** [comment]
I think I've renamed reasonably following your recommendations (I was still being somewhat HEAD centric, but if you are thinking of HEAD as opt-in that's fine).  Should these have inline comments about why each package is added or subtracted?  And just a comment block at the top to describe what each is supposed to test?

**Comment by jeanconn** [comment]
I think I've renamed reasonably following your recommendations (I was still being somewhat HEAD centric, but if you are thinking of HEAD as opt-in that's fine).  Should these have inline comments about why each package is added or subtracted?  And just a comment block at the top to describe what each is supposed to test?

**Comment by taldcroft** [comment]
`SKA3` test spec and/or package tests need work.  I know some of this is known / in-progress, but I got a lot of fails on Mac:
```
**************************************************************
***      Package                  Script            Status ***
*** ------------------ ---------------------------- ------ ***
***   Chandra.Maneuver                 test_unit.py   pass ***
***   Chandra.Maneuver           post_check_logs.py   pass ***
***       Chandra.Time                 test_unit.py   pass ***
***       Chandra.Time           post_check_logs.py   pass ***
*** Chandra.cmd_states                 test_unit.py   FAIL ***
*** Chandra.cmd_states           post_check_logs.py   FAIL ***
***         Quaternion                 test_unit.py   pass ***
***         Quaternion           post_check_logs.py   pass ***
***            Ska.DBI                 test_unit.py   pass ***
***            Ska.DBI           post_check_logs.py   pass ***
***          Ska.Numpy                 test_unit.py   pass ***
***          Ska.Numpy           post_check_logs.py   pass ***
***        Ska.ParseCM             test_unit_git.sh   FAIL ***
***        Ska.ParseCM           post_check_logs.py   FAIL ***
***          Ska.Shell                 test_unit.py   FAIL ***
***          Ska.Shell           post_check_logs.py   FAIL ***
***          Ska.Table             test_unit_git.sh   ---- ***
***          Ska.Table           post_check_logs.py   ---- ***
***     Ska.engarchive    test_make_archive_long.sh   ---- ***
***     Ska.engarchive                 test_unit.py   pass ***
***     Ska.engarchive           post_check_logs.py   pass ***
***            Ska.ftp                 test_unit.py   FAIL ***
***            Ska.ftp           post_check_logs.py   FAIL ***
***       Ska.quatutil             test_unit_git.sh   pass ***
***       Ska.quatutil           post_check_logs.py   pass ***
***            Ska.tdb                 test_unit.py   pass ***
***            Ska.tdb           post_check_logs.py   pass ***
***          acis_taco                 test_unit.py   pass ***
***          acis_taco           post_check_logs.py   pass ***
***       acisfp_check              test_regress.sh   FAIL ***
***       acisfp_check         test_regress_head.sh   ---- ***
***       acisfp_check           post_check_logs.py   FAIL ***
***       acisfp_check              post_regress.py   FAIL ***
***       acisfp_check         post_regress_head.py   ---- ***
***              agasc                 test_unit.py   pass ***
***              agasc           post_check_logs.py   pass ***
***              annie                 test_unit.py   FAIL ***
***              annie           post_check_logs.py   FAIL ***
***        chandra_aca                 test_unit.py   FAIL ***
***        chandra_aca           post_check_logs.py   FAIL ***
***            cxotime                 test_unit.py   pass ***
***            cxotime           post_check_logs.py   pass ***
***          dea_check              test_regress.sh   FAIL ***
***          dea_check         test_regress_head.sh   ---- ***
***          dea_check           post_check_logs.py   FAIL ***
***          dea_check              post_regress.py   FAIL ***
***          dea_check         post_regress_head.py   ---- ***
***          dpa_check              test_regress.sh   FAIL ***
***          dpa_check         test_regress_head.sh   ---- ***
***          dpa_check           post_check_logs.py   FAIL ***
***          dpa_check              post_regress.py   FAIL ***
***          dpa_check         post_regress_head.py   ---- ***
***               kadi         test_regress_long.sh   ---- ***
***               kadi                 test_unit.py   pass ***
***               kadi           post_check_logs.py   FAIL ***
***               kadi         post_regress_long.py   ---- ***
***              maude                 test_unit.py   pass ***
***              maude           post_check_logs.py   pass ***
***               mica                 test_unit.py   pass ***
***               mica           post_check_logs.py   pass ***
***   package_manifest              test_regress.sh   pass ***
***   package_manifest           post_check_logs.py   pass ***
***   package_manifest              post_regress.py   pass ***
***           parse_cm                 test_unit.py   pass ***
***           parse_cm           post_check_logs.py   pass ***
***            proseco                 test_unit.py   pass ***
***            proseco           post_check_logs.py   pass ***
***         psmc_check              test_regress.sh   FAIL ***
***         psmc_check         test_regress_head.sh   ---- ***
***         psmc_check           post_check_logs.py   FAIL ***
***         psmc_check              post_regress.py   FAIL ***
***         psmc_check         post_regress_head.py   ---- ***
***             pyyaks                 test_unit.py   pass ***
***             pyyaks           post_check_logs.py   pass ***
***           sparkles                 test_unit.py   pass ***
***           sparkles           post_check_logs.py   pass ***
***          starcheck              test_regress.sh   FAIL ***
***          starcheck           post_check_logs.py   pass ***
***          starcheck              post_regress.py   FAIL ***
***          timelines test_unit_git_sybase_long.sh   ---- ***
***          timelines           post_check_logs.py   ---- ***
***               xija                 test_unit.py   FAIL ***
***               xija           post_check_logs.py   FAIL ***
**************************************************************
```
The kadi fail is strange because it passes tests with `python setup.py test` or `import kadi; kadi.test` or `pytest kadi` but `import testr; testr.testr()` fails:
```
In [1]: import testr

In [2]: testr.testr()
======================================================================= test session starts ========================================================================
platform darwin -- Python 3.6.2, pytest-3.2.1, py-1.4.34, pluggy-0.4.0 -- /Users/aldcroft/miniconda3/envs/ska3/bin/python
cachedir: .cache
rootdir: /Users/aldcroft/git/kadi, inifile:
plugins: remotedata-0.3.0, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collecting 49 items                                                                                                                                                 Exception ignored in: <bound method SFTP.__del__ of <Ska.ftp.ftp.SFTP object at 0x119bc2160>>
Traceback (most recent call last):
  File "/Users/aldcroft/miniconda3/envs/ska3/lib/python3.6/site-packages/Ska.ftp-3.5-py3.6.egg/Ska/ftp/ftp.py", line 84, in __del__
    self.ftp.close()
  File "/Users/aldcroft/miniconda3/envs/ska3/lib/python3.6/site-packages/Ska.ftp-3.5-py3.6.egg/Ska/ftp/ftp.py", line 183, in __getattr__
    raise AttributeError("'{}' attr missing from Ska.ftp object".format(attr))
AttributeError: 'ftp' attr missing from Ska.ftp object
collected 53 items                                                                                                                                                  

kadi/cmds/tests/test_commands.py::test_find PASSED
```

**Comment by jeanconn** [comment]
Do you have the /proj/sot/ska/data/ska_testr data on the test machine?  And I think we hadn't decided if that was going into the default ska_sync or if "testers / package maintainers" could get that on their own.

Besides that, I think this PR might be the wrong place to track the test failures, unless you think those tests should be skipped on the test system.

**Comment by jeanconn** [comment]
Regarding the kadi failure, I think https://github.com/sot/Ska.ftp/pull/14 could clean up the ftp stuff but it has been a while since I've looked at it.  We might also be able to ignore the Ska.ftp weirdness from the ska_testr side if needed.

**Comment by jeanconn** [comment]
Also, it is really a fundamental issue that I'm not sure how tests should behave with an incomplete Ska.engarchive.   I just opened it as a Chandra.Maneuver issue athttps://github.com/sot/Chandra.Maneuver/issues/12 but I think it is the issue for a bunch of these tests.

**Pull request merged**